components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import styled, { css } from 'styled-components'
+
+// Common components
+import { SettingsText } from '../../../components/default'
+import { Toggle } from '../../../components/toggle'
+
+// Assets
+import InfoIcon from '../../../components/default/settings/assets/info-icon.svg'
+
+// Utilities
+import { getLocale } from '../../../../common/locale'
+
+const isDarkTheme = (p: any) => {
```

Comment:
Wow the theme type is so awkward :laughing: I had a look to see if we could use it instead but noooope :laughing:

Reply:
optional:
```suggestion
const isDarkTheme = (p: any) => p.theme.name === 'Brave Dark'
```

Reply:
Can't use ITheme from brave-ui? But either way we should not detect dark mode this way. We can simply use window.matchMedia if we must use JS for dark mode changes.

Reply:
Ooops, I copied this from https://github.com/brave/brave-core/blob/621d6fdd3b2dba4add21179a542cf6c1b5cddbbb/components/brave_new_tab_ui/components/default/settings/index.ts#L30 without any second thought. If this is not favorable, I guess I can replace this with `@media (prefers-color-scheme)` ?

Reply:
Oh, the only usecase of this function can be replaced with `var(--background1)`. Can be remoed.